pkgs/by-name/di/display3d/package.nix

```diff
@@ -0,0 +1,25 @@
+{
+  lib,
+  rustPlatform,
+  fetchCrate,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "display3d";
+  version = "0.1.17";
+
+  src = fetchCrate {
+    inherit pname version;
+    hash = "sha256-H5mynKJ6LY1+Izl9JfQQnStdDr1Z2TlA2r7XAXNNgpA=";
+  };
+
+  cargoHash = "sha256-vqjNgQ4xYduI3BczPnpNqFxNWkQhdBsCdLLAUIjxHic=";
+
+  meta = {
```

Comment:
I'm not sure if it's correct(BC i havn't seem anyone use fromCrate)
but I guess this should belong to fromSource

try to add 
`sourceProvenance = with lib.sourceTypes; [ fromSource ];`
https://nixos.org/manual/nixpkgs/unstable/#lib.sourceTypes.fromSource

Reply:
`display3d`'s github source currently doesn't have a Cargo.lock, but the crates.io version does, so only fetchCrate works at the moment

[`xq` also uses fetchCrate](https://github.com/NixOS/nixpkgs/blob/275baaeee3b73087ef3de602c583e0b48399f551/pkgs/by-name/xq/xq/package.nix), that's the package.nix that i used as a template. `xq`'s source is also available on github, which leads me to understand that it doesn't matter where the source is pulled from